internal/controllers/nmc_reconciler.go

```diff
@@ -155,6 +151,14 @@ func (r *NMCReconciler) Reconcile(ctx context.Context, req reconcile.Request) (r
 		delete(statusMap, moduleNameKey)
 	}
 
+	// removing label of loaded kmods
+	if !r.nodeAPI.IsNodeSchedulable(&node, nil) {
```

Comment:
This is a problematic conditions: it means that we ignore the tolerations and the cordoned node will be always unschedulable, and that means that the code never reaches line 165, which will handle unloading of the kernel module as needed during upgrade

Reply:
That depends on how upgrade is managed. We support 2 flows:
1. the label on the node is changed to a new version, and in that case the unloading will be done in the ProcessModuleSpec
2. the label on the node is removed, and will be later changed (this will allows user to execute whatever maintenance on the node that is needed). In that case the spec will be removed from NMC, and this scenario is handled in the ProcessUnconfiguredModuleStatus

Reply:
Thanks @yevgeny-shnaidman. In that case, let me move the API just before GarbageCollectInUseLabels(). That should help both the scenarios. Let me know if works

Reply:
@yevgeny-shnaidman Pls review this change as well